samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -365,6 +366,44 @@ class JobModelManager(
     }
   }
 
+  private def populateContainerLocalityMappings(): Unit = {
```

Comment:
1. I think the agreed upon solution in offline-discussion with @prateekm  was to manage the lifecycle of `LocalityManager` in `ClusterBaseJobCoordinator` and inject it in `JobModelManager` when generating the `JobModel`. Why was that route not taken?  
2. There're test failures( `TestSamzaYarnAppMasterService`) in CI. Does this patch break any existing functionality or is it simply un-updated mocks/tests?

Reply:
1. The ContainerProcessManager.java needs a handle to the localityManager to query containers' hosts, when handling container failures. 
So moving the LocalityManager to JC means, the CPM needs to access the LM via the JC. The JC is the one that creates and manages the CPM, thus creating a circular dependency. 

2. I have to talk to someone who knows about the SamzaYarnAppMasterService to investigate why the test is failing. The code-path being tested doesnt seem to be used in production.

Reply:
> Circular dependency
1.  If I understand right, ClusterBasedJobCoordinator is at one level higher than both the JobModelManager, ContainerProcessManager.  So how does managing the LocalityManager lifecycle above both  the JobModelManager, ContainerProcessManager in ClusterBasedJC and injecting it down during instantiation/jobModelGeneration would create circular dependency? Can you shed some light.
2. Does both these locality methods logically belong in JobModelManager? It would make the refactoring easier, but we have a  full blown out LocalityManager just to manage locality. Moving these locality computation outside of it breaks the ownership boundaries. 

> SamzaYarnAppMasterService:

Would be great to fix the existing  tests and test this end to end just to ensure that we're not breaking anything. Can you  share some light on the integration tests we did for this change?

>  code-path being tested doesnt seem to be used in production.

Are you really sure of that? AFAIK, it sets up ApplicationMasterWebServlet to list the config, appState, rmHttpAddress etc. It might not be a good idea to unnecessarily remove an endpoint as a part of this change.